src/Nethermind/Nethermind.Synchronization/BeamSync/BeamBlockchainProcessor.cs

```diff
@@ -260,18 +261,26 @@ private void BeamProcess(Block block)
                 _recoveryStep.RecoverData(block);
                 (IBlockchainProcessor beamProcessor, IStateReader stateReader) = CreateProcessor(block, new ReadOnlyDbProvider(_readOnlyDbProvider, true), _specProvider, _logManager);
 
-                BlockHeader parentHeader = _readOnlyBlockTree.FindHeader(block.ParentHash, BlockTreeLookupOptions.TotalDifficultyNotNeeded);
+                BlockHeader? parentHeader = _readOnlyBlockTree.FindHeader(block.ParentHash, BlockTreeLookupOptions.TotalDifficultyNotNeeded);
                 if (parentHeader != null)
                 {
-                    prefetchTasks = PrefetchNew(stateReader, block, parentHeader.StateRoot, parentHeader.Author ?? parentHeader.Beneficiary);
+                    if (parentHeader.StateRoot == null)
+                    {
+                        throw new InvalidDataException($"Received a block with null {nameof(parentHeader.StateRoot)} for beam processing");
+                    }
```

Comment:
When nulls would have thrown an NRE, there are a few options:
1. Do nothing and just let it throw an NRE.
2. make the error a bit more specific and give some better error messaging.
3. Change the datastructure so this field cannot be null.

(3) felt like it was likely to turn into a huge thing, and may not be desired.
(1) usually isn't desirable.
So I went with (2).

Reply:
agreed, part of the nullability work is adding more meaningful exceptions when you discover a proper null problem, bigger piece of pleasant work

generally Micah if you wish to fix the nullables in the entire codebase this is something that I would happily contract someone for - it would improve the codebase a lot

Reply:
I just want the warnings to go away while I'm writing my plugin.  😬  This is why I'm poor.